src/librustdoc/clean/mod.rs

```diff
@@ -2101,8 +2101,8 @@ impl Clean<Item> for doctree::Constant<'_> {
 impl Clean<Mutability> for hir::Mutability {
     fn clean(&self, _: &DocContext<'_>) -> Mutability {
         match self {
-            &hir::Mutability::Mutable => Mutable,
```

Comment:
why has rustdoc its own mutability enum?

Reply:
Something rustdoc-wants-to-be-more-independent something something, cc @GuillaumeGomez

Reply:
We do it for all hir types. If it can be removed, everyone will be happy though. ;)

Reply:
Don't see why it couldn't be removed; should be quite easy.

Reply:
I didn't look (and still haven't) so I don't know if it's possible or not. The idea behind `clean` rustdoc types is to provide a type with direct access to what rustdoc needs. If the rustdoc type doesn't provide anything more than the original, then it can (and should) be removed. Might be worth opening an issue about this.